### ISSUE TITLE
shutdown delay with 0.2.5

### ISSUE THREAD

**Issue by cjw296** [jruere/multiprocessing-logging#13]
I think there may be a regression in 0.2.5, which seems to wait for 5s on all shutdown for me.

[This code](https://github.com/jruere/multiprocessing-logging/blob/7fd2dcd74fe4a1f0f024ba4daa8be464bee66297/multiprocessing_logging.py#L66) should probably not execute on the receiving side based on [the documentation](https://docs.python.org/3.6/library/multiprocessing.html#multiprocessing.Queue.join_thread); indeed the docs suggest it's not needed on the sending side either, since it's the default behaviour.

For reference, here's how logbook does it:
https://github.com/getlogbook/logbook/blob/9d81786f505a5edabc14cfb2b7af6567c219e38e/logbook/queues.py#L493
...but there's no decent example to show how to wire those two together!

My guess is that there's some thread management issue lurking in multiprocessing-logging or there was a process being managed badly. Thoughts?

**Comment by jruere** [comment]
It was a bit more complicated. The thread which gets LogRecords from the queue, did an eternally blocking get so when the MultiprocessingLogger was closed, it would not realize it should stop unless it received a LogRecord.
The 5s timeout is from [here](https://github.com/jruere/multiprocessing-logging/blob/master/multiprocessing_logging.py#L98), where it waits for the thread to join.

I have introduced a timeout on the get so that it loops every 200ms.
Now the maximum delay is still 5s but only if you have many messages to log. This is a good thing.
The maximum wait when there are no messages is 200ms. I hope it's good enough.
